### PR TITLE
feat: Adds support for hidden input types

### PR DESCRIPTION
**Pull request by dhudec**
## Description

- Renames KeyboardType to InputType and adds support for `textPassword` and `numberPassword` input types to hide values and replace with dots
- Fixes [this](https://github.com/Basis-Theory/basistheory-android/security/dependabot/1) dependabot security alert by pinning to latest version of `gradle-build-action`

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

image

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: lib/src/main/java/com/basistheory/android/model/InputType.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.basistheory.android.model
+
+import android.text.InputType as AndroidInputType
+
+enum class InputType(val androidInputType: Int, val isConcealed: Boolean = false) {
+    TEXT(AndroidInputType.TYPE_CLASS_TEXT),
+    NUMBER(AndroidInputType.TYPE_CLASS_NUMBER),
+    TEXT_PASSWORD(AndroidInputType.TYPE_TEXT_VARIATION_PASSWORD, true),
+    NUMBER_PASSWORD(AndroidInputType.TYPE_CLASS_NUMBER, true);
```
Comment (dhudec) [inline lib/src/main/java/com/basistheory/android/model/InputType.kt:9]: Note: the android input type `TYPE_NUMBER_VARIATION_PASSWORD` displays a keyboard with alpha characters for some reason - this uses the number input type to display the correct virtual keyboard but conceals the value with a `TransformationMethod`

File: lib/src/main/java/com/basistheory/android/model/InputType.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.basistheory.android.model
+
+import android.text.InputType as AndroidInputType
+
+enum class InputType(val androidInputType: Int, val isConcealed: Boolean = false) {
```
Comment (jleon15) [inline lib/src/main/java/com/basistheory/android/model/InputType.kt:5]: there's a bunch of others we are not supporting, i guess we will only if ppl ask for them?

File: lib/src/main/java/com/basistheory/android/view/method/FullyHiddenTransformationMethod.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * Customizes PasswordTransformationMethod to also hide the current character while typing
+ */
+internal class FullyHiddenTransformationMethod : PasswordTransformationMethod() {
+
+    override fun getTransformation(charSequence: CharSequence, view: View): CharSequence {
+        return FullyHiddenCharSequence(charSequence)
+    }
+
+    private inner class FullyHiddenCharSequence(private val charSequence: CharSequence) : CharSequence {
+        private val dot = '\u2022'
```
Comment (jleon15) [inline lib/src/main/java/com/basistheory/android/view/method/FullyHiddenTransformationMethod.kt:16]: you can move this out as a const right? or within a companion object

**Review by jleon15 — APPROVED** [review]

File: lib/src/main/java/com/basistheory/android/model/InputType.kt (hunk 2 above)
Comment (dhudec) [reply lib/src/main/java/com/basistheory/android/model/InputType.kt:5]: Yeah, we were only going to support `textPassword` so even `numberPassword` is a bonus

**Pull request merged**